QUARKUS-1404.md

```diff
@@ -0,0 +1,29 @@
+# QUARKUS-1404 - Vert.x update to 4.2.x or 4.3.x
+
+JIRA link: https://issues.redhat.com/browse/QUARKUS-1404
+
+This test plan cover Vertx upgrade to 4.2.2
+
+## Related test plans
+
+ - [Quarkus-190](Quarkus-190)
+
+## Scope of the testing
+
+Test development will focus on
```

Comment:
What features in Quarkus does move 4.1 to 4.2 enables? 

Checked https://vertx.io/blog/whats-new-in-vert-x-4-2/ and looked for Quarkus blog posts, but didn't find anything super exciting.

Vertx HTTP-Client is Vert.x specific topic and there is Vert.x product to cover that. We should focus on Quarkus specifics.

Reply:
The integration that Quarkus does about Vertx HttpClient is done through this dependency:

```
<dependency>
  <groupId>io.smallrye.reactive</groupId>
  <artifactId>smallrye-mutiny-vertx-web-client</artifactId>
</dependency>
```

The features that I see that Vertx 4.2 brings are the following ones:

1. [Refactor the HTTP client sharing to manage client sharing with Vertx](https://github.com/eclipse-vertx/vert.x/pull/4190) or [Shared http client](https://github.com/eclipse-vertx/vert.x/pull/4172)

However, the way that Quarkus create an HTTPClient doesn't allow you to have a sharedHttpClient

Quarkus Way:
```
    @Inject
    Vertx vertx;
    
    private WebClient client;

    @PostConstruct
    void initialize() {
        // this.client = vertx.createSharedHttpClient(); not supported by Quarkus
        this.client = WebClient.create(vertx);
    }
```

So from a Quarkus Point of view, I think that this feature is not interesting. 

2. [WebSocket support](https://github.com/eclipse-vertx/vertx-http-proxy/pull/27)

Officially Quarkus use the following dependency in order to expose an endpoint over Websockets
```
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-websockets</artifactId>
</dependency>
```

 Vertx.4.2 upgrade and refactor WebSocket implementation, I am not sure if this point should be covered because is not the "documented" way to expose WebSockets.  WDYT ?

Reply:
At the end we concluded that we are not gonna need test development for this feature